### ISSUE TITLE
Gas masks don't work

### ISSUE THREAD

**Issue by ektoplazm** [CleverRaven/Cataclysm-DDA#40894]
# Describe the bug

The problem described in #40765 still persists, it's not possible to reload gas masks with cartridges (tested on gas mask, survivor gas mask, light survivor gas mask, XL gas mask), thus rendering them impossible to activate. However, testing it on a gas mask which I had loaded and activated beforehand (in build 10614), I found out that this mask stays activated and works in gas clouds without any charges loaded. 

# Steps To Reproduce

1. Spawn a gas mask
2. Spawn gas mask cartridges
3. Try to reload the mask
4. Get the "you can' reload the mask" message in the log. 
5. Get an activated gas mask (with no charges)
6. Spawn a bloated zombie
7. Walk into a gas cloud while wearing this activated gas mask with no charges, gas poisoning does not happen, no messages about "inhaling toxic gas" appear in the log. 

# Expected behavior

Gas masks can be reloaded with cartridges and consume them as charges when activated and used. 

# Screenshots

![image](https://user-images.githubusercontent.com/60448096/83041393-17c40300-a049-11ea-8ce5-2d7f266f2685.png)

# Versions and configuration

- OS: Windows
    - OS Version: 10.0 1909
- Game Version: 0.E-2532-gf4bb830 [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food]
]

# Additional context
this error occurs when trying to activate an empty light survivor mask - 

```
DEBUG    : can't set ammo null as it is not an ammo
FUNCTION : item& item::ammo_set(const itype_id&, int)
FILE     : src/item.cpp
LINE     : 583
```

**Comment by anothersimulacrum** [comment]
Just needs pocket data updating:
```diff
diff --git a/data/json/items/tool_armor.json b/data/json/items/tool_armor.json
index e4c6123184..6dd005062d 100644
--- a/data/json/items/tool_armor.json
+++ b/data/json/items/tool_armor.json
@@ -1192,9 +1192,7 @@
     "material_thickness": 2,
     "environmental_protection": 1,
     "environmental_protection_with_filter": 16,
-    "max_charges": 100,
-    "initial_charges": 100,
-    "ammo": "gasfilter_m",
+    "pocket_data": [ { "pocket_type": "MAGAZINE", "ammo_restriction": { "gasfilter_m": 100 } } ],
     "use_action": "GASMASK"
   },
   {
```

**Comment by wapcaplet** [comment]
This is fixed in #40997
